A tutorial blog on how to deploy meshery on AKS (#2965)
layer5io/layer5: src/collections/blog/2022/2022-07-07-deploy-meshery-on-aks/index.mdx

---
title: "How to deploy Meshery on AKS"
date: 2022-07-07 08:00:00 -0630
author: Srinivas Karnati
category: meshery
description: "How to deploy Meshery on Azure Kubernetes service(AKS)."
tags:
 - Meshery
 - Kubernetes
thumbnail: ./Meshery-on-AKS.png
type: Blog
published: true
resource: true
---

import { BlogWrapper } from "../../Blog.style.js";
import mesheryui from "./mesheryui.png";

<BlogWrapper>

> <p><em>Meshery is the open-source, cloud native management plane that can provision ten different service meshes, onboard your applications, manage WebAssembly filters, apply service mesh patterns, validate against best practices, and benchmarks the performance of your service mesh deployments.</em></p>

Reviewer: Link to `/meshery`

> <p><em><a href="https://meshery.io/">Meshery</a>  is the open-source, cloud native management plane that can provision ten different service meshes, onboard your applications, manage WebAssembly filters, apply service mesh patterns, validate against best practices, and benchmarks the performance of your service mesh deployments.</em></p>
> <p><em>Meshery’s vision is to make the operation of cloud native infrastructure and the service mesh layer of cloud simplified. Originally created by Layer5, Meshery is an open source project with hundreds of contributors world-wide and is actively maintained by engineers from Red Hat, VMware, Intel, Layer5 and others. </em></p>

Reviewer 2: ```suggestion
<div class="intro">
<p><a href="https://meshery.io/">Meshery</a>'s goal is to make the operation of cloud native infrastructure and the service mesh layer of cloud simplified. Originally created by Layer5, Meshery is an open source project with hundreds of contributors world-wide and is actively maintained by engineers from Red Hat, VMware, Intel, Layer5 and others.</p>
</div>
```

<h2>Setup and run Meshery on AKS</h2>
<p>The following instructions expects you to have an active Azure subscription, and Azure CLI installed on your system. </p>
<h3> Spin up the AKS Cluster</h3>
<p>Create the resource group (a logical group where all our resources will be deployed). The following command creates  a resource group named MesheryGroup in <code>southindia</code> location. </p>
<pre><code class="language-bash">
az group create --name MesheryGroup --location southindia
</code></pre>

<p>Create AKS cluster using <code>az aks create</code>. The following command creates aks cluster with a single node. </p>

<pre><code class="language-bash">
az aks create --resource-group MesheryGroup --name MesheryAKS --node-count 1 --generate-ssh-keys
</code></pre>
<p>After a few minutes, the command completes and returns a JSON formatted information about the cluster.</p>

> <p>You can connect with your cluster by using <em>az aks get-credentials</em> ,  which basically downloads credentials and configure the Kubernetes CLI. </p>

Reviewer: ```suggestion
<p>You can connect with your cluster by using <code>az aks get-credentials</code> ,  which basically downloads credentials and configure the Kubernetes CLI. </p>
```

<pre><code>
az aks get-credentials --resource-group MesheryGroup --name MesheryAKS
</code></pre>

> <p>Verify the connection to your cluster using the kubectl get command. </p>

Reviewer: ```suggestion
<p>Verify the connection to your cluster using the <code>kubectl get</code> command. </p>
```
